File: clases 25-11-2025/tipos de clases Java/clase normal con interfaz/Curso.java
public class Curso implements ICurso {
    public String nombre;
    public int duracion;

    // Constructor sin parámetros
    public Curso() {
        this.nombre = "sin nombre";
        this.duracion = 0 ;
    }

    @Override
    public void infoCurso() {
        System.out.println("Curso: " + nombre + ", Duración: " + duracion + " horas");
    }
}

File: clases 25-11-2025/Tipos de clases js/clase normal con interfaz/Clases/Curso.js
import ICurso from '../interfaces/ICurso.js';

export default class Curso extends ICurso {
    constructor() {
        super();
        this.nombre = "sin nombre";
        this.duracion = 0;
    }

    infoCurso() {
        console.log(`Curso: ${this.nombre}, Duración: ${this.duracion} horas`);
    }
}